### PR TITLE
fix(server): optimized commit download

### PR DESCRIPTION
**Pull request by fabis94**

### CODE REVIEW

**Comment by gjedlicska** [comment]
I'd like to add a test, that ensures, we can insert a very big number of closures, to prevent from regressions.

File: packages/server/modules/core/services/objects.js [hunk 1]
```diff
@@ -50,7 +50,11 @@ module.exports = {
     await Objects().insert(insertionObject).onConflict().ignore()
 
     if (closures.length > 0) {
-      await Closures().insert(closures).onConflict().ignore()
+      const batchSize = 10000
+      while (closures.length > 0) {
```
[the remark this answers is not in the document]
Comment (fabis94) [reply packages/server/modules/core/services/objects.js:54]: @gjedlicska I intentionally didn't use it out of concerns for memory usage. It requires you to pass in the entire array and I don't know how optimally it splices that array up to ensure there's no duplication in memory

[the remark this answers is not in the document]
Comment (fabis94) [reply packages/server/modules/core/services/objects.js:54]: Oh and also - no way to add onConflict().ignore()

[the remark this answers is not in the document]
Comment (gjedlicska) [reply packages/server/modules/core/services/objects.js:54]: yeah, that rings a bell.

lets just add a test then

**Review by gjedlicska — APPROVED** [review]

**Pull request merged**